EDIINT Chartering (#59)
OR13/diem-draft-charter: ediint/README.md

# EDIINT Working Group Charter

> The EDIINT Working Group is chartered to update and modernize the mechanisms defined in RFC 4130 (AS2 – Applicability Statement 2) to reflect advances in Internet transport security, messaging reliability, and interoperability practices, while maintaining backward compatibility with existing implementations where practical.

Reviewer: ```suggestion
The EDIINT Working Group is chartered to update the mechanisms defined in RFC 4130 (MIME-Based Secure Peer-to-Peer Business Data Interchange Using HTTP, Applicability Statement 2 (AS2)) to reflect advances in Internet transport security, messaging reliability, and interoperability practices, while maintaining backward compatibility with existing implementations where practical.
```

Reviewer: * "update ....to reflect advences.." is what I guess is meant by "modernize".  I don't think "modernize" adds much here.
* Better to include the full document  title.

## Background

> The original EDIINT Working Group developed protocols enabling secure and reliable business data interchange over the Internet, including AS1, AS2, and AS3. These technologies have been widely adopted in global commerce. Despite the availability of alternative technologies such as RESTful interfaces over HTTPS, AS2 remains foundational in numerous production environments and industry frameworks. Many sectors continue to depend on AS2 due to its proven reliability, strong authentication and non-repudiation capabilities, and established certification and compliance ecosystems. This update focuses on modernizing AS2 to maintain its operational relevance and interoperability in today's Internet environment.

Reviewer: ```suggestion
The EDIINT Working Group developed protocols enabling secure and reliable business data interchange over the Internet, including AS1 (RFC3335), AS2 (RFC4130), and AS3 (RFC4823). These applicability statements have been widely adopted. Despite the availability of alternative technologies such as RESTful interfaces over HTTPS, AS2 remains foundational in numerous production environments and industry frameworks. Many sectors continue to depend on AS2 due to its proven reliability, strong authentication and non-repudiation capabilities, and established certification and compliance ecosystems. This update focuses on refreshing AS2 to maintain its operational relevance and interoperability in today's Internet environment.
```

* As ediint is expected to be revived. We can get rid of "original".
* I suggest we simply delete "in global commerce". If you keep it, then please consider rewording it as I don't parse what is meant there.
* Remove "modernizing" to avoid the apparent contradiction with not supporting "RESTful interfaces"

## Goals and Scope

The Working Group will:

> 1. **Modernize RFC 4130 (AS2):**

Reviewer: ```suggestion
1. **Update RFC 4130 (AS2) to reflect recent practices and recommendations:**
```

> - Align with current Internet and security practices (e.g., TLS 1.3, modern cipher suites, certificate validation, MIME updates).

Reviewer: ```suggestion
- Align with current Internet and security practices (e.g., TLS 1.3, recommended cipher suites, certificate validation, MIME updates).
```

> - Replace outdated dependencies with references to current standards and best practices.

Reviewer: ```suggestion
- Replace outdated dependencies with references to latest standards and best practices.
```

- Improve clarity in message structure, signatures, receipts, and error handling.

2. **Clarify Backward Compatibility Expectations:**

- Define guidance to ensure existing AS2 deployments continue operating without disruption, while recognizing that some legacy mechanisms—whether cryptographic, transport, configuration, or feature-related—may not interoperate with updated implementations.
- Clarify that backward compatibility focuses on preserving operational continuity and migration pathways, not guaranteeing interoperability across all legacy features or security levels.
- Where older algorithms or mechanisms are retained solely for compatibility, they will be designated appropriately relative to their modern alternatives.

3. **Clarify and Extend Functionality:**

- Incorporate operational experience and lessons from two decades of AS2 deployment.

> - Address implementation ambiguities, interoperability challenges, and deployment best practices.

Reviewer: ```suggestion
- Address implementation ambiguities and interoperability challenges
- Document deployment best practices.
```

- Consider optional, informational extensions that improve operational usability or integration without redefining AS2 as a new protocol.

4. **Deliverables**

> - A standards-track document updating RFC 4130 ("RFC 4130bis").

Reviewer: ```suggestion
- A Standards Track document updating RFC 4130 ("RFC 4130bis").
```

> - Optional informational drafts describing interoperability and transition strategies, covering best practices and implementation profiles.

Reviewer: ```suggestion
- Optional informational documents describing interoperability and transition strategies, covering best practices and implementation profiles.
```

> - Optional informational documents describing interoperability and transition strategies, covering best practices and implementation profiles.

Reviewer: ```suggestion
- Optional Informational documents describing interoperability and transition strategies, covering best practices and implementation profiles.
```

## Out of Scope

- Development of new EDI or transport protocols (e.g., REST/HTTPS-based EDI messaging).
- Changes that break compatibility with existing AS2 deployments beyond what may be required for security or standards alignment.

> 

Reviewer 2: - AS1 and AS3 will not be addressed as part of this work.

Author: applied here: https://github.com/ietf-artarea/charters/pull/59#discussion_r2589601679

> 

Author: ```suggestion
- AS1 and AS3 will not be addressed as part of this work.
```

Reviewer 2: Change the comma to a period.  That was a typo.

## Milestones

- **WG Adoption of RFC 4130bis draft as Proposed Standard**: Initial working group draft (Q2 2026)
- **WG Last Call**: Consensus on updates and backward compatibility sections (Q4 2026)

> - **IETF Last Call / Publication Request**: Submission to IESG for publication (Q2 2027)

Reviewer: ```suggestion
- **Publication Request**: Submission to IESG for publication (Q2 2027)
```

IETF LC is not part of state controlled by WGs.

> - **WG Last Call**: Consensus on updates and backward compatibility sections (Q4 2026)

Reviewer: ```suggestion
- **WG Last Call** of the  document RFC4130bis document (Q4 2026)
```

- Clarify that backward compatibility focuses on preserving operational continuity and migration pathways, not guaranteeing interoperability across all legacy features or security levels.
- Where older algorithms or mechanisms are retained solely for compatibility, they will be designated appropriately relative to their modern alternatives.

3. **Clarify and Extend Functionality:**

- Incorporate operational experience and lessons from two decades of AS2 deployment.
- Address implementation ambiguities and interoperability challenges
- Document deployment best practices.
- Consider optional, informational extensions that improve operational usability or integration without redefining AS2 as a new protocol.

4. **Deliverables**

- A Standards Track document updating RFC 4130 ("RFC 4130bis").
- Optional Informational documents describing interoperability and transition strategies, covering best practices and implementation profiles.

## Out of Scope

- Development of new EDI or transport protocols (e.g., REST/HTTPS-based EDI messaging).
- Changes that break compatibility with existing AS2 deployments beyond what may be required for security or standards alignment.

> - AS1 and AS3 will not be addressed as part of this work.

Author: ```suggestion
- AS1 and AS3 will not be addressed as part of this work.
- Updating or obsoleting RFC 6017 and RFC 6362 will not be addressed as part of this work.
```

Reviewer: This is clear about the scope.

If there is no plan to absorb these features in the bis, then maybe:

"Updating or obsoleting RFC 6017 and RFC 6362 will not be addressed as part of this work."

Leave it to you to decide where to put the cursor.
